src/components/lib/IFrame.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+import { classNames } from "@/utils";
+import React, { useEffect, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+
+interface IFrameProps {
+  className?: string;
+  children: React.ReactElement;
+}
+
+const IFrame = ({ children, className = "" }: IFrameProps) => {
+  const ref = useRef<HTMLIFrameElement>(null);
+  const [iframeDoc, setIframeDoc] = useState<Document>();
+  const [loadBody, setLoadBody] = useState(false);
+
+  useEffect(() => {
+    setIframeDoc(ref.current?.contentWindow?.document);
+  }, []);
+
+  useEffect(() => {
+    const stylesheet = document.querySelector<HTMLLinkElement>(
```

Comment:
Is the stylesheet we're storing coming from the DevUI (the parent project's) stylesheet? 

If that's the case, it could potentially lead to templates being affected and conflicting with the custom styles of the parent project in the development environment.

We've placed the templates in an iframe to ensure that they are completely isolated from the parent DevUI project. 

I'd appreciate hearing your thoughts on this to confirm if my understanding is correct.

Reply:
Yes the `stylesheet` is coming from DevUI stylesheet.

Yes, I agree with you that it could lead to templates being affected by parent's custom styles. But since we are using only `tailwind css` in templates, browser would pick only tailwind css from the parent's styles and apply them to templates. We know custom styles won't ever be used in templates so it won't affect them.

However, it surely breaks our purpose of completely isolating templates from DevUI. To keep it isolated, we might use `runtime tailwind css generator` and append styles in `iframe` which will lead to unnecessary load time and increased bundle size. I am calling it `unnecessary` because if we have already generated css somewhere then why do we need to generate again for every template.

alternatively, we can use `tailwind cdn` but it will ultimately kill HRM (hot-reload) feature in the `development` environment since it requires repainting the template which replaces react components with html code and HRM does not work with html code.

I would love to hear if you have any better solution so that we can optimise it more further.

Reply:
@wajeshubham @deepesh-dg Based on my understanding of the context here is one suggestion I would like to share regarding the leakage of CSS from parent to the iframe. The iframe can utilise shadow dom.

I'm not sure of the possibility of it being added on the project but just adding my suggestion for a specific problem discussed here.